src/main/java/com/vcgdev/demo/config/auth/RestTemplateInterceptor.java

```diff
@@ -0,0 +1,36 @@
+package com.vcgdev.demo.config.auth;
+
+import java.io.IOException;
+
+import org.springframework.http.HttpRequest;
+import org.springframework.http.client.ClientHttpRequestExecution;
+import org.springframework.http.client.ClientHttpRequestInterceptor;
+import org.springframework.http.client.ClientHttpResponse;
+import org.springframework.security.oauth2.common.OAuth2AccessToken;
+import org.springframework.security.oauth2.provider.authentication.OAuth2AuthenticationDetails;
+import org.springframework.security.oauth2.provider.token.TokenStore;
+
+public class RestTemplateInterceptor implements ClientHttpRequestInterceptor {
+
+    private IAuthenticationFacade authFacade;
+    private TokenStore tokenStore;
+
+    public RestTemplateInterceptor(IAuthenticationFacade authFacade, TokenStore tokenStore) {
+        this.authFacade = authFacade;
+        this.tokenStore = tokenStore;
+    }
+
+    @Override
+    public ClientHttpResponse intercept(HttpRequest request, byte[] body, ClientHttpRequestExecution execution)
+            throws IOException {
+        if (!request.getHeaders().containsKey("Authorization")) {
+            Object details = authFacade.getAuthentication().getDetails();
+            if (details instanceof OAuth2AuthenticationDetails) {
+                OAuth2AuthenticationDetails oauthDetails = (OAuth2AuthenticationDetails) details;
+                OAuth2AccessToken token = tokenStore.readAccessToken(oauthDetails.getTokenValue());
```

Comment:
There is no access to token store  from external microservices... use token from header assumption:
token type -> Bearer